engine/src/flutter/shell/platform/android/io/flutter/embedding/engine/systemchannels/SensitiveContentChannel.java

```diff
@@ -0,0 +1,191 @@
+              }
+              break;
+            case "SensitiveContent.getContentSensitivity":
+              try {
+                final Integer currentContentSensitvity =
+                    sensitiveContentMethodHandler.getContentSensitivity();
+                result.success(serializeContentSensitivity(currentContentSensitvity));
+              } catch (IllegalStateException | IllegalArgumentException e) {
+                result.error("error", e.getMessage(), null);
+              }
+              break;
+            case "SensitiveContent.isSupported":
+              result.success(sensitiveContentMethodHandler.isSupported());
+              break;
+            default:
+              Log.v(
+                  TAG, "Method " + method + " is not implemented for the SensitiveContentChannel.");
+              result.notImplemented();
+              break;
+          }
+        }
+      };
+
+  // The annotation to suppress the "InlinedApi" lint prevents lint warnings
+  // caused by usage of Android 35 APIs. This call is safe because these values
+  // will never be used to set content sensitivity for a View via a
+  // SensitiveContentMethodHandler because it must ensure a device is API 35+ before
+  // doing so to avoid exceptions on other devices.
+  @SuppressLint({"InlinedApi"})
+  private int deserializeContentSensitivity(String contentSensitivityName) {
+    switch (contentSensitivityName) {
+      case AUTO_SENSITIVE_CONTENT_SENSITIVITY:
+        return View.CONTENT_SENSITIVITY_AUTO;
+      case SENSITIVE_CONTENT_SENSITIVITY:
+        return View.CONTENT_SENSITIVITY_SENSITIVE;
+      case NOT_SENSITIVE_CONTENT_SENSITIVITY:
+        return View.CONTENT_SENSITIVITY_NOT_SENSITIVE;
+      default:
+        throw new IllegalArgumentException(
+            "contentSensitivityName "
+                + contentSensitivityName
+                + " not known to the SensitiveContentChannel.");
+    }
+  }
+
+  // The annotation to suppress the "InlinedApi" lint prevents lint warnings
+  // caused by usage of Android 35 APIs. This call is safe because these values
+  // will never be used to set content sensitivity for a View via a
+  // SensitiveContentMethodHandler because it must ensure a device is API 35+ before
+  // doing so to avoid exceptions on other devices.
+  @SuppressLint({"InlinedApi"})
+  private String serializeContentSensitivity(int contentSensitivityValue) {
+    switch (contentSensitivityValue) {
+      case View.CONTENT_SENSITIVITY_AUTO:
+        return AUTO_SENSITIVE_CONTENT_SENSITIVITY;
+      case View.CONTENT_SENSITIVITY_SENSITIVE:
+        return SENSITIVE_CONTENT_SENSITIVITY;
+      case View.CONTENT_SENSITIVITY_NOT_SENSITIVE:
+        return NOT_SENSITIVE_CONTENT_SENSITIVITY;
+      default:
+        // Signal to Flutter framework that the embedder does not recognize
+        // the content sensitivity mode queried.
+        return UNKNOWN_CONTENT_SENSITIVITY;
```

Comment:
shouldn't this return error result instead?

Reply:
No. The idea here was that it a user calls `getContentSensitivity` and an unknown content sensitivity is returned by Android, we should not crash because the user may have no idea that an unknown one was returned, e.g. if we bump to Android 16 and the default content sensitivity changes. Instead, we return unknown to signal to the framework side to handle this accordingly.

I'm sure we could throw an error to accomplish the same thing but I think using the unknown value is a bit clearer in terms of our intent.

Reply:
sgtm.